compress.hh

```diff
@@ -91,28 +99,32 @@ public:
     static const sstring CHUNK_LENGTH_KB_ERR;
     static const sstring CRC_CHECK_CHANCE;
 private:
-    compressor_ptr _compressor;
+    algorithm _algorithm;
```

Comment:
I think that std::optional\<algorithm> captures the different between no compression better (but it is really a minor distinction).